scripts/vaccination_alerts/send-emails.ts

```
@@ -0,0 +1,206 @@
+        await this.onInvalidEmail(email);
+      } else {
+        console.error(
+          `Error sending email ${email}, ${alert.fips} - ${alert.emailAlertVersion}.`,
+          err,
+        );
+        this.errorCount += 1;
+      }
+    }
+  }
+}
+
+/**
+ * Given a list of users to email and a list of locations, email those
+ * users with alert information about those locations.
+ *
+ * Takes in a file with {[fips: string]: {...data}} and file like {[fips: string]: [emails]}
+ *
+ * Also takes in a parameter to actual send the email. Otherwise will just generate the files
+ *
+ * You can run via `yarn send-emails fipsToAlertFilename currentSnapshot [send]`
+ */
+async function main(alertPath: string, dryRun: boolean, singleEmail?: string) {
+  const alertsByFips = readVaccinationAlerts(alertPath);
+  const alerts = _.values(alertsByFips);
+
+  const emailService = new EmailService();
+  const db = getFirestore();
+
+  const alertEmailService = new SendVaccinationAlertsService(db, emailService);
+
+  const emailAlertTuples = _.flatten(
+    await Promise.all(
+      alerts.map(async alert => {
+        const emails = await alertEmailService.fetchEmailsForAlert(alert);
+        return (
+          emails
+            // if an email is specified, filter out all other emails. Used for testing
+            .filter(email => (singleEmail ? email === singleEmail : true))
+            .map(email => ({ email, alert }))
+        );
+      }),
+    ),
+  );
+
+  const emailBatches = _.chunk(emailAlertTuples, BATCH_SIZE);
+  console.log(
+    `Sending ${emailAlertTuples.length} emails in ${emailBatches.length} batches.`,
+  );
+
+  for (const batch of emailBatches) {
+    await Promise.all(
+      batch.map((data: { email: string; alert: RegionVaccinePhaseInfo }) =>
+        alertEmailService.sendAlertEmail(data.email, data.alert, dryRun),
+      ),
+    );
+  }
+
+  alertEmailService.logEmailsToBeSentStats();
+
+  if (dryRun) {
+    return;
+  }
```

It looks like you removed the code that checks `alertEmailService.errorCount`, but I think we need that.  If there are any errors, we should make sure the process fails with a non-zero exit code so that the workflow fails and we are notified that there was a problem.  Else it'll end up just silently failing.

Note that we explicitly handle invalid emails separately from `errorCount` so those won't cause a problem. https://github.com/covid-projections/covid-projections/pull/2711/files#diff-f89fbdcacb69b99011a18b990ed557c1c1481810027222a5e6ae761de4075e25R103

---

Adding this back!